### DISCUSSION TITLE
Enable scope but only for `function_declaration`

### DISCUSSION THREAD

**Discussion by serranomorante** [lukas-reineke/indent-blankline.nvim#826] [Q&A]
Hi, I'm only interested on **scope** for `function_declaration`. I tried to configure `ibl` like this (see snippet) but on javascript and typescript files ("javascript", "tsx", "typescript" parsers) it will still highlight scopes like `if_statement`.

So how can I only enable `function_declaration` for scope?

```lua
  "lukas-reineke/indent-blankline.nvim",
  main = "ibl",
  opts = {
    ...
    scope = {
      enabled = true,
      include = { node_type = { ["*"] = { "function_declaration" } } },
      show_start = false,
      show_end = false,
      exclude = {
        node_type = {
          ["*"] = { "chunk", "do_statement", "while_statement", "repeat_statement", "if_statement", "for_statement" },
        },
      },
    },
```

**Comment by lukas-reineke** [comment]
To exclude everything except functions, you would only need to exclude `statement_block`, `for_statement`, `for_in_statement`, and `catch_clause`.
And `jsx_element` if you are using react.

**Reply by serranomorante** [reply]
it works! thank you thank you so much 💖
